src/v/kafka/client/client.cc

```diff
@@ -278,15 +278,15 @@ client::create_topic(kafka::creatable_topic req) {
 }
 
 ss::future<list_offsets_response>
-client::list_offsets(model::topic_partition tp) {
+client::list_offsets(const model::topic_partition& tp) {
     return gated_retry_with_mitigation([this, tp]() {
         return _topic_cache.leader(tp)
           .then(
             [this](model::node_id node_id) { return _brokers.find(node_id); })
-          .then([tp](auto broker) mutable {
+          .then([tp](auto broker) {
               return broker->dispatch(kafka::list_offsets_request{
                 .data = {.topics{
-                  {{.name{std::move(tp.topic)},
+                  {{.name{tp.topic},
```

Comment:
what am i missing here? it looks like it was ok before since the move was from a captured copy.

Reply:
I agree. I don't know why this fixes it. I'd expect it to fail if it looked like:
```cpp
gated_retry_with_mitigation([this, tp]() mutable {
   something(std::move(tp));
```

Reply:
From my understanding the issue is the functor being called twice and `tp` being moved out on the first call , leaving it empty on subsequent invocations. That means the solution shouldn't need the const & addition in the function argument, but rather just to only omit the call to `std::move`

Reply:
if the lambda in here `.then([tp](auto broker) {` is being reused by the top-level lambda that is retried, that would make sense. we could probably verify that with some assertions or looking at how seastar works. in either case, i think we need an explanation for what's happening before merging.

Reply:
I have fixes incoming.

Reply:
@dotnwat yes thats whats going on due to the nature of `gated_retry_with_mitigation` the lambda that you pass in will be invoked again upon failure, after the error handler you pass in is invoked.

Reply:
right but the lambda that is being passed in isn't the lambda whose capture has the `tp` moved-from. it is the later on in the fiber that is _created_ when that lambda is re-run.

Reply:
> From my understanding the issue is the functor being called twice and `tp` being moved out on the first call , leaving it empty on subsequent invocations. That means the solution shouldn't need the const & addition in the function argument, but rather just to only omit the call to `std::move`

Yes. But that doesn't work.